composer.json

```diff
@@ -447,6 +447,9 @@
             },
             "drupal/tfa_email_otp": {
                 "Disable Verify button on first landing to the tfa form - https://www.drupal.org/project/tfa_email_otp/issues/3469826#comment-15740693": "https://www.drupal.org/files/issues/2024-08-23/3469826-disable-verify-button.patch"
+            },
+            "drupal/elasticsearch_connector": {
+                "Allow dashes in index name": "https://gist.githubusercontent.com/nicksantamaria/95677ffa87a9e1433bfc27c3ebdefd35/raw/68d6eb055150d1e99ad6f0da49d68a6a4c2d6348/elasticsearch_connector-dashed-index.patch"
```

Comment:
We are already extending `IndexFactory` here `TideSearchIndexFactory` and overriding few other methods. So I think it is better to override this function here and update references in `create` function here `$indexName = IndexFactory::getIndexName($index);`

Reply:
hmm good call @vincent-gao 
I didn't realise it was getting called in multiple places. I was just looking into the override function we have and saw it has only one call. I am happy to use this patch for now as we are still using the old version of the module, there is no point of submitting a patch to that.